components/dialog/RotateEncryptionKeyDialog.vue

```diff
@@ -24,27 +26,53 @@ export default {
   },
 
   async fetch() {
-    const c = this.cluster;
+    if (!this.$store.getters['isRancher']) {
+      // This fetch function is getting snapshots, which are associated
+      // with cluster manager. Rancher Desktop doesn't come with cluster
+      // manager, so for Rancher Desktop we return nothing.
+      return [];
+    }
 
-    if ( c.isRke1 && this.$store.getters['isRancher'] ) {
-      let etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
+    const getEtcdBackups = async() => {
```

Comment:
Maybe we can get this outside from fetch?

Reply:
Moved it into `methods`